oak/server/rust/oak_runtime/src/io.rs

```diff
@@ -89,6 +89,9 @@ impl<T: Decodable> Receiver<T> {
                     })
                 })
                 .and_then(|message| T::decode(&message))
+        } else if read_status[0] == ChannelReadStatus::Orphaned {
+            info!("Channel closed {:?}", self.handle);
+            Err(OakStatus::ErrChannelClosed)
```

Comment:
I am not a big fan of mapping channel statuses to oak statuses, when the underlying abi actually returned something else. I think @rbehjati and I also discussed this in the past. I don't have a concrete suggestion, but it feels like we should try to reduce this friction.

For instance, Streams would return `Option<T>`, in which `None` means the stream is exhausted: https://docs.rs/futures/0.3.5/futures/stream/trait.Stream.html

Perhaps a similar pattern would help here?

Reply:
Yeah, the use of `OakStatus` on Runtime-internal methods is convenient for when they get mapped to/from ABI functions, but definitely induces friction elsewhere.  Leaving for now as it's independent of this PR.